lib/rules/UseEffectRulePlugin.js

```diff
@@ -77,19 +116,16 @@ class UseEffectRulePlugin {
 								if (!ident) ident = defaultIdent;
 								references.set(ident, options);
 							}
-							return {
-								type: enforce ? `use-${enforce}` : "use",
-								value: {
-									loader,
-									options,
-									ident
-								}
-							};
+							return ruleSetCompiler.createEffect(type, {
+								loader,
+								options,
+								ident
+							});
 						}
 					};
 
 					/**
-					 * @param {any} items user provided use value
+					 * @param {Array<string|UseRule>|string|UseRule} items user provided use value
```

Comment:
Looks like `useToEffectsWithoutIdent ` does not support `items` as function https://github.com/webpack/webpack/blob/next/declarations/WebpackOptions.d.ts#L193 . So in Typescript definition it is possible to define 
```
{
     "use": () => [() => {loader: 'url-loader'}]
}
```
See https://github.com/webpack/webpack/blob/next/declarations/WebpackOptions.d.ts#L185
cc @sokra

Reply:
As well as https://github.com/webpack/webpack/blob/next/declarations/WebpackOptions.d.ts#L185
https://github.com/webpack/webpack/blob/next/declarations/WebpackOptions.d.ts#L193 are invariant